wasmplugin/plugin.go

```diff
@@ -100,8 +95,34 @@ func (ctx *corazaPlugin) OnPluginStart(pluginConfigurationSize int) types.OnPlug
 		return types.OnPluginStartStatusFailed
 	}
 
+	directivesAuthoritiesMap := map[string][]string{}
+	for authority, directivesName := range config.perAuthorityDirectives {
+		if _, ok := directivesAuthoritiesMap[directivesName]; !ok {
+			directivesAuthoritiesMap[directivesName] = []string{}
+		}
+
+		directivesAuthoritiesMap[directivesName] = append(directivesAuthoritiesMap[directivesName], authority)
+	}
+
 	perAuthorityWAFs := newWAFMap(len(config.directivesMap))
 	for name, directives := range config.directivesMap {
+		var (
+			authorities   []string
+			setDefaultWAF bool
```

Comment:
If I'm not mistaken, an empty authorities slice can be used instead of this boolean. It seems to make sense code-wise too below for the usage, "if I don't have authorities, use default, otherwise use the authorities"

Reply:
Codewise empty, not nil, like `if len(authorities) == 0` is quite natural isn't it? The idea came to mind because the current variable looked sort of random.

Reply:
i am afraid the intention of the code isn't reflecting the logic. Basically here we coallesce the default waf with per-authority waf setup to avoid code duplication. `len(authorities) == 0` in here does not mean `no authorities and hence use default` it means `authorities have been set, hence we are in the case of building the default WAF`. All this avoidance of code duplication comes from the fact that if two authorities use the same WAF we don't want to spawn two instance of the same WAF, instead reuse it and hence the logic of setting up a WAF for an authority vs setting up a WAF for the default WAF are similar.

Reply:
Hmm still don't quite follow the logic, I thought the default waf is the one that doesn't have authorities associated with it so is precisely what's used when there are no authorities in this parse. Are you saying there is a case to use default waf even when there are authorities - present? I don't see it from the code but could be missing it.

But if so, then probably code sharing can still happen by extracting another function or two. I think it should be possible for setupDefaultWaf and another function to exist instead of the now even weirder variable name, while still sharing actual business logic between the two.

Reply:
I am sorry for the confusion. What you say is true, when `isTheDefaultWAF == true`, `directives` will be always empty, what I am saying here is that that is incidental, `if len(directives) == 0` implying we should set that WAF as the default WAF isn't intuitive, it is more a consequence that they happen in the same group of code and understanding why that waf with no directives associated should be the default one requires you to go back up and read the code again, whereas the condition:

```go
if isTheDefaultWAF {
    perAuthorityWAFs.setDefaultWAF(waf)
}
```

is straightward.

Reply:
> you to go back up and read the code again, whereas the condition

For me it was the exact opposite having to go back up to understand what this variable actually means. I think at a minimum then the loop needs to move to an else block to make the relationship between boolean and slice clearer. I really think the variable isn't helping much at that point though ;)